### DISCUSSION TITLE
Can ModelViewer load more than one gltf mode in one scene ？

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#4067] [Q&A] [answered]
I use below code to lode multiple models，but only the last is show.
Can some one help with giving some suggestion.
Thanks !

` private fun createRenderables() {
        val buffer = assets.open("models/A.gltf").use { input ->
            val bytes = ByteArray(input.available())
            input.read(bytes)
            ByteBuffer.wrap(bytes)
        }
        modelViewer.loadModelGltfAsync(buffer) { uri -> readCompressedAsset("models/$uri") }

        val buffer2 = assets.open("models/B.gltf").use { input ->
            val bytes = ByteArray(input.available())
            input.read(bytes)
            ByteBuffer.wrap(bytes)
        }
        modelViewer.loadModelGltfAsync(buffer2) { uri -> readCompressedAsset("models/$uri") }}

        modelViewer.transformToUnitCube()

    }`

**Comment by romainguy** [comment, marked as answer]
`ModelViewer` is a helper that only shows one model at a time. If you want to manage more complex scenes, you should use AssetLoader directly. You can check the [source code](https://github.com/google/filament/blob/main/android/filament-utils-android/src/main/java/com/google/android/filament/utils/ModelViewer.kt) of `ModelViewer` to see how to do it (just copy/paste and modify as needed).

**Reply by fanbinnet** [reply]
Change ModelViewer as your suggestion,It work fine now.
Thank you very much.

**Reply by Wheats** [reply]
Hello, can you tell me how to load multiple models，

i hava three models.  and I need to distinguish three different click events,

**Comment by Wheats** [comment]
Hello, can you tell me how to load multiple models，

i hava three models. and I need to distinguish three different click events,

**Reply by fanbinnet** [reply]
In ModelViewer  add asset2,asset3 and add other code refer the code used for  "asset" .

**Reply by Wheats** [reply]
fun loadModelGlb1(buffer: Buffer) {
        destroyModel()
        asset1 = assetLoader.createAssetFromBinary(buffer)
        asset1?.let { asset ->
            resourceLoader.asyncBeginLoad(asset)
            animator = asset.animator
            asset.releaseSourceData()
        }
    }

A new FilamentAsset has been added.
But none of them are displayed

**Comment by Wheats** [comment]
1.
resourceLoader.asyncBeginLoad(asset)
 replace
resourceLoader.loadResources(asset)

2.transformToUnitCube
 add asset1 setTransform

3.destroyModel
 add asset1 destroyAsset

4.render
 asset1?.let { populateScene(it) }


will done

3Q

**Reply by ttyomaa6** [reply]
Hello, have you had any problems with the transformation of models, it just doesn't work quite the way I need it
